src/Task/Phpcs.php

```diff
@@ -161,7 +164,76 @@ private function addArgumentsFromConfig(
         $arguments->addOptionalCommaSeparatedArgument('--ignore=%s', $config['ignore_patterns']);
         $arguments->addOptionalCommaSeparatedArgument('--exclude=%s', $config['exclude']);
         $arguments->addOptionalArgument('-s', $config['show_sniffs_error_path']);
+        $arguments->addOptionalArgument('--parallel=%s', $this->getParallelValue($config['parallel']));
 
         return $arguments;
     }
+
+    protected function getParallelValue(string|int|null $option): ?int
+    {
+        if ($option === null) {
+            return null;
+        }
+
+        if (is_string($option)) {
+            if ($option === 'auto') {
```

Comment:
To keep the maintenance of this codebase manageable, we rely on the tools to provide the options for us.
Does phpcs have a way to automatically detect the number of cores? If not, it should probably be done in their rather than in this package.

Ideally, the implementation fo passing the cores options, should be as simple as:
`$arguments->addOptionalArgument('--parallel=%s', $config['parallel']);`

Relying on phpcs to do the proper validations of the int range / auto value - where null means full back to phpcs defaults.


Additional note here, is that GrumPHP itself also is running in parallel and possibly other tasks are running in parallel as well. If they all take up all cores, it might have an impact on the system.